### PR TITLE
chore: upgrade various packages to remove jwt-go v3.2.0 dependency

### LINKED ISSUE #1186: Security Vulnerability found in this Package.
https://github.com/Azure/aad-pod-identity/blob/a5afff63eb14f0baa3507cbcdb02594cb5f55d0f/go.sum#L112

We are using aad-pod-identity package and our security application reported vulnerability in this package. Is there any plan to upgrade this package?

https://nvd.nist.gov/vuln/detail/CVE-2020-26160Vulnerable package: github.com/dgrijalva/jwt-go
fixed in v4.0.0-preview1

### PR DESCRIPTION
**Pull request by chewong**
Signed-off-by: GitHub <noreply@github.com>

**Reason for Change**:

Fixes #1186 

**Requirements**

- [x] squashed commits
- [ ] included documentation
- [ ] added unit tests and e2e tests (if applicable). See [test standard](https://github.com/Azure/aad-pod-identity/blob/master/CONTRIBUTING.md#test-standard) for more details.
- [ ] ran `make precommit`

**Issue Fixed**:

**Please answer the following questions with yes/no**:

Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

- [ ] yes
- [ ] no

**Notes for Reviewers**:

### CODE REVIEW

File: test/e2e/go.mod [hunk 1]
```diff
@@ -5,66 +5,67 @@ go 1.17
 require (
 	github.com/Azure/aad-pod-identity v1.8.2
```
Comment (aramase) [inline test/e2e/go.mod:6]: Lets also bump this to `v1.8.5`

File: test/e2e/go.mod [hunk 2]
```diff
@@ -5,66 +5,67 @@ go 1.17
...
-	k8s.io/api v0.22.1
-	k8s.io/apiextensions-apiserver v0.22.1
-	k8s.io/apimachinery v0.22.1
-	k8s.io/client-go v0.22.1
-	sigs.k8s.io/controller-runtime v0.10.0
+	k8s.io/api v0.23.0
+	k8s.io/apiextensions-apiserver v0.23.0
+	k8s.io/apimachinery v0.23.0
+	k8s.io/client-go v0.23.0
+	sigs.k8s.io/controller-runtime v0.11.0-beta.0.0.20211208212546-f236f0345ad2
```
Comment (aramase) [inline test/e2e/go.mod:21]: Do we need to use the beta release or could we just use the latest stable one `v0.10.3`?

**Review by aramase — COMMENTED** [review]
Could you also rebase the PR?

File: test/e2e/go.mod (hunk 2 above)
Comment (chewong) [reply test/e2e/go.mod:21]: v0.10.3 won't work because it is still using Kubenretes v1.22. Added a TODO comment to upgrade to v0.11.0.

**Review by aramase — APPROVED** [review]

**Pull request merged**